File: solutions/0169-majority-element/solution.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = defaultdict(int)
        n = len(nums)

        for num in nums:
            count[num] += 1
            if count[num] > n/2:
                return num
        

File: solutions/0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int val =0, cnt =0;
        for(int i : nums){
            if(cnt == 0) val = i;
            if(i == val) cnt++;
            else cnt--;
        }
        return val;
    }
};
